core/src/com/unciv/logic/city/CityConstructions.kt

```diff
@@ -469,6 +456,41 @@ class CityConstructions {
                     builtBuildingUniqueMap.addUnique(unique)
     }
 
+    fun addFreeBuildings() {
+        // "Gain a free [buildingName] [cityFilter]"
+        val freeBuildingUniques = cityInfo.getLocalMatchingUniques(UniqueType.GainFreeBuildings, StateForConditionals(cityInfo.civInfo, cityInfo))
+
+        for (unique in freeBuildingUniques) {
+            val freeBuildingName = cityInfo.civInfo.getEquivalentBuilding(unique.params[0]).name
+            val citiesThatApply = when (unique.params[1]) {
+                "in this city" -> listOf(cityInfo)
+                "in other cities" -> cityInfo.civInfo.cities.filter { it !== cityInfo }
+                else -> cityInfo.civInfo.cities.filter { it.matchesFilter(unique.params[1]) }
+            }
+
+            for (city in citiesThatApply) {
+                if (city.cityConstructions.containsBuildingOrEquivalent(freeBuildingName)) continue
+                city.cityConstructions.addBuilding(freeBuildingName)
+                if (city.id !in freeBuildingsProvidedFromThisCity)
+                    freeBuildingsProvidedFromThisCity[city.id] = hashSetOf()
+
+                freeBuildingsProvidedFromThisCity[city.id]!!.add(freeBuildingName)
+            }
+        }
+
+        // Civ-level uniques - for these only add free buildings from each city to itself to avoid weirdness on city conquest
+        for (unique in cityInfo.civInfo.getMatchingUniques(UniqueType.GainFreeBuildings, stateForConditionals = StateForConditionals(cityInfo.civInfo, cityInfo))) {
+            val freeBuildingName = cityInfo.civInfo.getEquivalentBuilding(unique.params[0]).name
+            if (cityInfo.matchesFilter(unique.params[1])) {
+                if (cityInfo.id !in freeBuildingsProvidedFromThisCity)
```

Comment:
These three lines are exactly the same as above, not sure if it's worth separating tbh

Reply:
:rage:  f*ing github just ate a detailed answer 'cuz I tried to avoid typing the filename and copy it. Lesson learned: _never_ compose text in the browser, use a notepad++ tab. Summary: Nope, I just moved the function as it violated the (in this file) still mostly correct regions (now correct). You know, a state-changing function in the pure region. Which I still didn't touch in MapUnit - a mess. I should learn that tenet I just postulated and keep my grubby fingers off such finds.

Any functional cleanup of that (`getLocalMatchingUniques` / `getMatchingUniques` / `cityInfo.matchesFilter` ignores "in this city" and "in other cities") mess needs **_testing_** which is not the scope of this PR. By someone who understands the architecture, which (seeing that) I'm not. Pity git blame is unable to track that relocation, I would need to switch branches to see whodunit.

So, do I revert the region cleanup or leave it - no, by now clearer to leave it in.